pkg/controllers/clusterresourcebindingwatcher/watcher.go

```diff
@@ -104,16 +110,24 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func isGenerationUpdated(oldBinding, newBinding *fleetv1beta1.ClusterResourceBinding) bool {
-	return oldBinding.Generation != newBinding.Generation
+func areConditionsUpdated(oldBinding, newBinding *fleetv1beta1.ClusterResourceBinding) bool {
+	for i := condition.RolloutStartedCondition; i < condition.TotalCondition; i++ {
+		oldCond := oldBinding.GetCondition(string(i.ResourceBindingConditionType()))
+		newCond := newBinding.GetCondition(string(i.ResourceBindingConditionType()))
+		if oldCond == nil && newCond == nil {
+			break
+		}
+		if isConditionUpdated(oldCond, newCond) {
+			return true
+		}
+	}
+	return false
 }
 
-func isConditionUpdated(oldCondition, newCondition *metav1.Condition) bool {
-	if oldCondition == nil && newCondition == nil {
-		return false
-	}
-	if oldCondition == nil || newCondition == nil {
+func isConditionUpdated(oldCond, newCond *metav1.Condition) bool {
+	if oldCond == nil || newCond == nil {
 		return true
 	}
-	return oldCondition.Status != newCondition.Status
+	// We don't compare message because in general it's changed every time reason is changed. We also don't check lastTransitionTime for similar reason.
```

Comment:
the comment about lastTransitionTime is not correct. we don't care about the lastTransitionTime field at all.